edx/event-routing-backends
# Ability to set per router configuration

**Description:** New fields to be added to RouterConfiguration 1. Router url  2. active and removed ConfigurationModel support in our model. Add new migrations for new fields. Add with modifications ConfigurationModel functions in RouterConfiguration

**JIRA:** https://edlyio.atlassian.net/browse/ERTE-1

**Dependencies:** No dependencies 
**Merge deadline:** None

**Installation instructions:** Run lms migrations

**Testing instructions:**

1. Open page http://localhost:18000/admin/event_routing_backends/routerconfiguration/
2. Add  Router Configuration 
Backend Name(xapi/caliper)
url: take url form https://webhook.site for example
Sample configuration for caliper https://webhook.site/ae76a382-5eed-4ae7-8b0f-7dfe650482be
[{"match_params":{"context.host":"localhost:18000"},"router_type":"AUTH_HEADERS","host_configurations":{"url":"https://webhook.site/ae76a382-5eed-4ae7-8b0f-7dfe650482be"}}]
Sample configuration for xapi
[{"match_params":{"context.host":"localhost:18000"},"router_type":"XAPI_LRS","host_configurations":{"url":"https://webhook.site/ae76a382-5eed-4ae7-8b0f-7dfe650482be"}}]

3. Enroll/Un enroll any course. Check if event is post on given url from https://webhook.site
4. Able to add multiple Router Configuration from admin and active/deactivate them

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.
